### PR TITLE
MTA: Check mandatories incidents before reporting

### PR DESCRIPTION
**Pull request by gabriel-farache**
Closes [FLPATH-1422](https://issues.redhat.com/browse/FLPATH-1422)

MTA: Check mandatories incidents before reporting and advising for M2K

### CODE REVIEW

File: mta-v6.x/specs/notifications.json [hunk 1]
```diff
@@ -12,9 +12,7 @@
       }
     ],
     "servers": [
-      {
-        "url": "http://localhost:7007/api/notifications"
-      }
+
```
Comment (masayag) [inline mta-v6.x/specs/notifications.json:15]: I'd prefer aligning the mtav6 and mtav7 with mta and use the notifications in yaml format, preferably, using the extension to include it from the same reference.

Comment (masayag) [reply mta-v6.x/specs/notifications.json:15]: having said that, I'm going to override this part entirely with the PR to switch to the new notifications API.

Comment (masayag) [reply mta-v6.x/specs/notifications.json:15]: I guess I'll have to take care for it during the rebase after merging this PR.

File: mta-v7.x/argfile.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+BUILDER_IMAGE=brew.registry.redhat.io/rh-osbs/openshift-serverless-1-logic-swf-builder-rhel8@sha256:012439f21f964478225a459042f6e1dc14f63fdf4ae089c4ee5e613814682a27
+FLOW_NAME=MTA Serverless Workflow
```
Comment (masayag) [inline mta-v7.x/argfile.conf:2]: pls specify v7 in the name

File: mta-v7.x/mta.sw.yaml [hunk 3]
```diff
@@ -55,6 +57,14 @@ functions:
   - name: logInfo
     type: custom
     operation: "sysout:INFO"
+  - name: countMandatoriesIncidents
+    type: expression
+    operation: .applicationAnalysis | [select (.issues[].category=="mandatory")][0] | [.issues[]?.incidents //[]] | .[-1][-1].id
+    # cat anal.json | jq '. | [select (.issues[].category=="mandatory")][0] | [.issues[]?.incidents //[]] | .[-1][-1].id'
```
Comment (masayag) [inline mta-v7.x/mta.sw.yaml:63]: should the comment remain or be removed?

File: mta-v7.x/mta.sw.yaml [hunk 4]
```diff
@@ -359,7 +420,7 @@ states:
           refName: createNotification
           arguments:
             title: "MTA analysis completed"
-            message: '"MTA analysis is completed and the report for " + .application.repository.url + " is ready."'
+            message: '"MTA analysis is completed with " + (.application.nonMandatoriesIncidents|tostring) + " non mandatories incidents and the report for " + .application.repository.url + " is ready."'
```
Comment (masayag) [inline mta-v7.x/mta.sw.yaml:423]: s/non mandatories../non-mandatory incidents. The report for " + .application.repository.url + " is ready."/

File: mta-v6.x/specs/notifications.json (hunk 1 above)
Comment (masayag) [reply mta-v6.x/specs/notifications.json:15]: (in other words, pls ignore those comments :-) )

File: mta-v7.x/mta.sw.yaml (hunk 3 above)
Comment (gabriel-farache) [reply mta-v7.x/mta.sw.yaml:63]: oh yes, thx

**Review by masayag — APPROVED** [review]

**Pull request merged**